quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -466,29 +465,35 @@ private static void handleDeprecated(Set<String> deprecatedInUse, PropertyMapper
             optionName = optionName.substring(NS_KEYCLOAK_PREFIX.length());
         }
 
-        StringBuilder sb = new StringBuilder("\t- ");
-        sb.append(optionName);
+        StringBuilder sb = new StringBuilder();
+        if (mapper.getDeprecatedMetadata().isPresent()) {
 
-        if (!deprecatedValuesInUse.isEmpty()) {
-            sb.append("=").append(String.join(",", deprecatedValuesInUse));
-        }
+            if (!mapper.getDeprecatedMetadata().get().getNewOptionsKeys().isEmpty() || mapper.getDeprecatedMetadata().get().getNote() != null || !mapper.getDeprecatedMetadata().get().getDeprecatedValues().isEmpty()) {
```

Comment:
I'm afraid we can't do this. With this condition that don't have any new keys, note, or value will not be printed. A warning should be emitted when e.g. Keycloak is started as:
```
kc.sh start-dev --features=hostname:v1 --hostname=abc
```